pkg/plugins/client.go

```diff
@@ -0,0 +1,32 @@
+package plugins
+
+import (
+	"errors"
+
+	"github.com/hashicorp/go-plugin"
+)
+
+var ErrClientAlreadyClosed = errors.New("client already closed")
+
+// Client wraps the go-plugin client (which is in fact a plugin controller) and the GRPC client.
+type Client struct {
+	client *plugin.Client
+	log    HClogger
+}
+
+// NewClient constructs a new Client instance
+func NewClient(client *plugin.Client, log HClogger) Client {
+	return Client{
+		client: client,
+		log:    log,
+	}
+}
+
+// Close terminates the plugin
+func (c Client) Close() error {
+	if c.client == nil {
+		return ErrClientAlreadyClosed
+	}
```

Comment:
Nit: Without locking this is thread-unsafe!  [Interface io.Closer](https://pkg.go.dev/io#Closer) says closing twice is undefined, and that implementations may document their own behaviour.  We should document that this Close may only be called once.  Additionally we would additionally ensure that it is thread-safe to rely on this error being returned.

Reply:
this code doesn't exist in the next PR